int-tests/src/test-integration/java/io/neow3j/protocol/Neow3jReadOnlyIntegrationTest.java

```diff
@@ -745,6 +745,30 @@ public void testInvokeFunctionWithTransfer() throws IOException {
         assertNotNull(invoc.getTx());
     }
 
+    @Test
+    public void testInvokeFunctionWithDiagnostics() throws IOException {
+        List<ContractParameter> params = Arrays.asList(
+                hash160(new Hash160(committeeAccountScriptHash())),
+                hash160(new Hash160(defaultAccountScriptHash())),
+                integer(1),
+                integer(1));
+        Signer signer = AccountSigner.calledByEntry(new Hash160(committeeAccountScriptHash()))
+                .setAllowedContracts(NEO_HASH);
+
+        InvocationResult invoc = getNeow3j()
+                .invokeFunctionDiagnostics(NEO_HASH, "transfer", params, signer)
+                .send()
+                .getInvocationResult();
+
+        assertNotNull(invoc);
+        assertNotNull(invoc.getScript());
+        assertThat(invoc.getState(), is(NeoVMStateType.HALT));
+        assertNotNull(invoc.getGasConsumed());
+        assertNull(invoc.getException());
+        assertTrue(invoc.getStack().get(0).getBoolean());
+        assertNotNull(invoc.getTx());
```

Comment:
I think a check for the diagnostics is missing here?

Reply:
You're correct, I'll add an assertion for it.